.github/workflows/ci.yml

```diff
@@ -0,0 +1,94 @@
+name: CI
+
+on:
+  push:
+  release:
+    types: [published]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest, macos-latest ]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup constants
+        shell: bash
+        run: |
+          TOIT_VERSION=latest
+          echo "TOIT_VERSION=$TOIT_VERSION" >> $GITHUB_ENV
+          export DOWNLOAD_DIR="${{ github.workspace }}/downloads"
+          echo "DOWNLOAD_DIR=$DOWNLOAD_DIR" >> $GITHUB_ENV
+          if [[ "$RUNNER_OS" = "Linux" ]]; then
+            TOIT_FILE=toit-linux.tar.gz
+            echo "TOIT_EXEC=$DOWNLOAD_DIR/toit/bin/toit.run" >> $GITHUB_ENV
+            echo "TPKG_EXEC=$DOWNLOAD_DIR/toit/bin/toit.pkg" >> $GITHUB_ENV
+          elif [[ "$RUNNER_OS" = "macOS" ]]; then
+            TOIT_FILE=toit-macos.tar.gz
+            echo "TOIT_EXEC=$DOWNLOAD_DIR/toit/bin/toit.run" >> $GITHUB_ENV
+            echo "TPKG_EXEC=$DOWNLOAD_DIR/toit/bin/toit.pkg" >> $GITHUB_ENV
+          elif [[ "$RUNNER_OS" = "Windows" ]]; then
+            TOIT_FILE=toit-windows.tar.gz
+            echo "TOIT_EXEC=$DOWNLOAD_DIR/toit/bin/toit.run.exe" >> $GITHUB_ENV
+            echo "TPKG_EXEC=$DOWNLOAD_DIR/toit/bin/toit.pkg.exe" >> $GITHUB_ENV
+          else
+            echo "UNSUPPORTED RUNNER: $RUNNER_OS"
+            exit 1
+          fi
+
+          if [[ $TOIT_VERSION = latest ]]; then
+            echo "TOIT_URL=https://github.com/toitlang/toit/releases/latest/download/$TOIT_FILE" >> $GITHUB_ENV
+          else
+            echo "TOIT_URL=https://github.com/toitlang/toit/releases/download/$TOIT_VERSION/$TOIT_FILE" >> $GITHUB_ENV
+          fi
+
+      # Fetch the dependencies. Different for each platform.
+      - name: Install dependencies - Linux
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt-get update
+          sudo apt-get install ninja-build
+          ninja --version
+          cmake --version
+      - name: Install dependencies - macOS
+        if: runner.os == 'macOS'
+        run: |
+          brew install ninja
+          ninja --version
+          cmake --version
+      - name: Install dependencies - Windows
+        if: runner.os == 'Windows'
+        run: |
+          choco install ninja
+          ninja --version
+          cmake --version
+
+      - uses: suisei-cn/actions-download-file@v1
+        name: Download Toit
+        with:
+          url: ${{ env.TOIT_URL }}
+          target: ${{ env.DOWNLOAD_DIR }}
+
+      - name: Extract Toit
+        shell: bash
+        run: |
+          cd "$DOWNLOAD_DIR"
+          tar x -f *.tar.gz
```

Comment:
Doesn't this need a -z flag

Reply:
not with recent 'tar' executables ("recent" being 10? years old).
The only time when one has to provide one, is when piping.
See https://www.gnu.org/software/tar/manual/html_node/gzip.html (search for "Reading compressed archive is even simpler").